### ISSUE TITLE
[bug] Metamask window is not opening after disconnect

### ISSUE THREAD

**Issue by josadcha** [rainbow-me/rainbowkit#2033]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### RainbowKit Version

2.1.1

### wagmi Version

2.9.9

### Current Behavior

After disconnecting from Metamask and doing an attempt to connect again, the Metamask window is not opened. Issue resolves after refreshing the page:

https://github.com/rainbow-me/rainbowkit/assets/39953699/6e6eda1f-44d6-4e6b-970f-308523fc02a2

### Expected Behavior

After disconnecting from Metamask and trying to initiate the connection again, the Metamask is triggered to be connected.

### Steps To Reproduce

1. Connect to Metamask
2. DIsconnect
3. Try to connect to Metamask again

(Sometimes the issue appears on the 2nd attempt, sometimes on the 3rd.

**Comment by magiziz** [comment]
@josadcha Looking into this now

**Comment by REPTILEHAUS** [comment]
+1 Im also getting this and with the same Wagmi / Rainbowkit versions

**Comment by magiziz** [comment]
I'm able to reproduce this as well. Reporting this to the wagmi team to see what's going on.

**Comment by ducls-2223** [comment]
+1 I'm also getting this with Wagmi/Rainbowkit version "latest"

**Comment by magiziz** [comment]
Seems like this is a EIP-1193 and EIP-6963 issue. We should be deprecating EIP-1193 pretty soon and fix this. I'll keep this issue opened meanwhile 👍

**Comment by babkenmes** [comment]
The problem is in disconnecting the wallet. 

it becomes obvious after debugging [`packages/rainbowkit/src/components/ConnectOptions/DesktopOptions.tsx`](https://github.com/rainbow-me/rainbowkit/blob/d81935cd1f736ab701c686ed105b38b13a6956f5/packages/rainbowkit/src/components/ConnectOptions/DesktopOptions.tsx#L114)

```
  const connectToWallet = (wallet: WalletConnector) => {
    setConnectionError(false);
    if (wallet.ready) {
      wallet?.connect?.()?.catch(() => {
        setConnectionError(true);
      });
    }
  };
  ```

`wallet.connect()` throws `ConnectorAlreadyConnectedError`

The solution is to disconnect via direct connector. To achieve this you can utilize `useAccount` from wagmi.

```
const { connector } = useAccount()
const handleDisconnect = ()=>{
        connector && await connector.disconnect()
}
```

**Comment by magiziz** [comment]
@babkenmes We have `metaMaskWallet` EIP-1193 and EIP-6963 connector which shares the same `window.ethereum` instance, that's why it throws that error since it's marked as already "connected" from Wagmi. 

Feel free to check out the wagmi issue here https://github.com/wevm/wagmi/issues/4027. The issue should be resolved once we deprecate the EIP-1193 connectors 👍

**Comment by magiziz** [comment]
Here is another EIP-1193 bug we've encountered https://github.com/wevm/wagmi/issues/3606. As i said all these EIP-1193 breaking behaviours should be resolved once we migrate fully over to EIP-6963.

**Comment by babkenmes** [comment]
I see now)). Thanks for the clarification. That will be great, if I understand it correctly duplicate connectors will also be removed.

**Comment by magiziz** [comment]
@josadcha Are you able to reproduce this with the latest Wagmi version `2.11.1` ?

**Comment by REPTILEHAUS** [comment]
any update on this, Ive downgraded but need the latest versions that contain balance update fixes now like this https://github.com/rainbow-me/rainbowkit/pull/1717

**Comment by REPTILEHAUS** [comment]
@magiziz I think this is resolved now using 
```
    "@rainbow-me/rainbowkit": "^2.1.3",
    "viem": "^2.17.4",
    "wagmi": "^2.11.1",
```

**Comment by magiziz** [comment]
@REPTILEHAUS Awesome. Yeah Wagmi has done a few changes to their connectors so hopefully we should see some less conflicts now.

**Comment by magiziz** [comment]
This is now fixed in the latest Wagmi version `2.11.3`.
